lib/BackbeatConsumer.js

```diff
@@ -902,17 +909,8 @@ class BackbeatConsumer extends EventEmitter {
             return setTimeout(() => this.close(cb), 1000);
         }
         this._circuitBreaker.stop();
-        return async.series([
+        return async.waterfall([
             next => {
-                if (this._consumer) {
-                    try {
-                        this._consumer.commit();
-                    } catch (e) {
-                        // Ignore exceptions if we are not connected
-                        const logger = this._consumer.isConnected() ? this._log.error : this._log.debug;
-                        logger.bind(this._log)('commit failed', { e: e.toString() });
-                    }
-                }
                 if (this._kafkaBacklogMetricsConfig) {
                     // publish offsets to zookeeper
                     return this._publishOffsetsCron(() => next());
```

Comment:
publishing offsets must be done _after_ the unsusbribe is complete.

Reply:
maybe an easy wait to get around this is to introduce a flag to skip `disconnect` on timeout during shutdown: this way we can perform the `unsubscribe`, and when unassign'ed (and commited) we can do the offset publishing..

Reply:
actually, maybe publishing to zookeeper should always be done?